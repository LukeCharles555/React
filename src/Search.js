import React, { Component } from "react";
import Axios from "axios";
import MovieList from "./MovieList";
import { Button } from 'react-bootstrap';
import { Navbar, Nav, Form, FormControl } from 'react-bootstrap';

export default class Search extends React.Component {
    constructor(props) {

        super(props);
    }
    state = {
        moviesList:[],
        searchTerm: ''
    };


    search = event => {
        event.preventDefault();
        Axios.get("http://www.omdbapi.com/?s=" + this.state.searchTerm + "&apikey=407b600b")
            .then(res => res.data)
            .then(res => {
                if(!res.Search) {
                    this.setState({moviesList: [] });
                    return;
                }

                const moviesList = res.Search.map(movie => movie.imdbID);
                this.setState({
                    moviesList
                });
            });
    };

    handleChange = event => {
        this.setState({
            searchTerm: event.target.value
        });
    };

    render() {
        const { moviesList } = this.state;

        return (
            <div>
                <Navbar className="navbar" bg="dark" expand="lg">
                    <Navbar.Brand href="#home">OMDB</Navbar.Brand>
                    <Navbar.Toggle aria-controls="basic-navbar-nav" />
                    <Navbar.Collapse id="basic-navbar-nav">
                        <Nav className="mr-auto">
                            <Nav.Link href="#home">Home</Nav.Link>
                            <Nav.Link href="#link">Link</Nav.Link>

                        </Nav>
                        <Form onSubmit={this.search} inline>
                            <FormControl type="text" placeholder="Search" className="mr-sm-2" onChange={this.handleChange} />
                            <Button type="submit" variant="outline-success" id="searchButton">Search</Button>
                        </Form>
                    </Navbar.Collapse>
                </Navbar>
                {moviesList.length > 0 ? (
                    moviesList.map(movie => (
                        <MovieList movieID={movie} key={movie} history={this.props.history}/>
                    ))
                ) : (
                    <p>Couldn't find a movie, try again</p>
                )}
    </div>


        );
    }
}