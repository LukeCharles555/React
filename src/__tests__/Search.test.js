import React from 'react';
import { render } from '@testing-library/react';
import Search from "../Search";

test("render a search", () => {
    const wrapper = render(<Search />);
    expect(wrapper.baseElement).toMatchSnapshot();
})