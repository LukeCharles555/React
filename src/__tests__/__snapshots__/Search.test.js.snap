// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a search 1`] = `
<body>
  <div>
    <div>
      <nav
        class="navbar navbar-expand-lg navbar-light bg-light"
      >
        <a
          class="navbar-brand"
          href="#home"
        >
          OMDB
        </a>
        <button
          aria-controls="basic-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="basic-navbar-nav"
        >
          <div
            class="mr-auto navbar-nav"
          >
            <a
              class="nav-link"
              data-rb-event-key="#home"
              href="#home"
            >
              Home
            </a>
            <a
              class="nav-link"
              data-rb-event-key="#link"
              href="#link"
            >
              Link
            </a>
          </div>
          <form
            class="form-inline"
          >
            <input
              class="mr-sm-2 form-control"
              placeholder="Search"
              type="text"
            />
            <button
              class="btn btn-outline-success"
              type="submit"
            >
              Search
            </button>
          </form>
        </div>
      </nav>
      <p>
        Couldn't find a movie, try again
      </p>
    </div>
  </div>
</body>
`;
